dot/state/epoch.go

```diff
@@ -573,24 +591,45 @@ func (s *EpochState) FinalizeBABENextEpochData(epoch uint64) error {
 // getting the set of hashes from the received epoch and for each hash
 // check if the header is in the database then it's been finalized and
 // thus we can also set the corresponding NextConfigData in the database
-func (s *EpochState) FinalizeBABENextConfigData(epoch uint64) error {
+func (s *EpochState) FinalizeBABENextConfigData(finalizedHeader *types.Header) error {
 	s.nextConfigDataLock.Lock()
 	defer s.nextConfigDataLock.Unlock()
 
-	finalizedNextConfigData, err := lookupForNextEpochPersistedHash(s.nextConfigData, s, epoch)
+	finalizedBlockEpoch, err := s.GetEpochForBlock(finalizedHeader)
 	if err != nil {
+		return fmt.Errorf("cannot get epoch for block %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	nextEpoch := finalizedBlockEpoch + 1
+
+	configInDatabase, err := s.getConfigDataInDatabase(nextEpoch)
+	if err != nil && !errors.Is(err, chaindb.ErrKeyNotFound) {
+		return fmt.Errorf("cannot check if next epoch config is already defined for epoch %d: %w", nextEpoch, err)
+	}
+
+	// config already defined we don't need to lookup in the map
+	if configInDatabase != nil {
+		return nil
+	}
+
+	// not every epoch will have `ConfigData`
+	finalizedNextConfigData, err := lookupForNextEpochPersistedHash(s.nextConfigData, s, nextEpoch)
```

Comment:
can we rename this function `lookupForNextEpochPersistedHash` to something like `findNextInfoFromMemory`?

Reply:
This function checks in the memory for the hash which is persisted in the database, what about `findFinalizedHeaderForEpoch`? meaning that we need to have in the epoch map one hash that is persisted in the database.

Reply:
the function doesn't return a header though so that's kind of confusing I think